File: backend/src/Shalom.Application/Contracts/PersonDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// A person in the circle. <paramref name="LastContactedOn"/> and
/// <paramref name="SnoozedUntil"/> are the only stored connection facts —
/// whether a nudge is due is always derived at read time (ADR-004).
/// </summary>
public record PersonDto(
    Guid Id,
    string Name,
    string? Relationship,
    string? Phone,
    int? ContactCadenceDays,
    string? Notes,
    DateOnly? LastContactedOn,
    DateOnly? SnoozedUntil);

File: frontend/projects/api/src/lib/models/person-dto.ts
/**
 * A person in the circle. `lastContactedOn` / `snoozedUntil` are
 * server-derived local days (`yyyy-MM-dd`, ADR-004) — whether a nudge is
 * due is always computed server-side, never stored or client-derived.
 */
export interface PersonDto {
  readonly id: string;
  readonly name: string;
  readonly relationship: string | null;
  readonly phone: string | null;
  readonly contactCadenceDays: number | null;
  readonly notes: string | null;
  readonly lastContactedOn: string | null;
  readonly snoozedUntil: string | null;
}
